src/app/util/CHIPDeviceCallbacksMgr.h

```diff
@@ -69,6 +70,7 @@ class DLL_EXPORT CHIPDeviceCallbacksMgr
         if (CHIP_NO_ERROR == err)
         {
             ca->Cancel();
+            CHIPPlatformMemoryFree(ca->mInfoPtr);
```

Comment:
How do we ensure that if CancelCallback is not called, memory is freed? What is the call path of the "success" scenario of the callback being called?

Reply:
+1 here - I have a hard time following and convincing myself that 'memory if freed correctly in all code paths'.

Here for example I expect to at least have a `ca->mInfoPtr = nullptr` assignment